src/climatevision/generator/transport2030/t.py

```diff
@@ -111,10 +111,11 @@ def calc(
                 transport2018=t18.t,
             ),
         )
-        assert (
-            required_domestic_transport_capacity_pkm
-            <= res.transport.transport_capacity_pkm
-        ), "We should know have at least as much provided transport capacity as we required initially"
+        # IS THIS NECESSARY? Not possible with updated traffic air/ship
```

Comment:
No this isn't "necessary" it was a sanity check. But what exactly do you mean by not possible with updated traffic air /ship?

Reply:
Also, das Thema ist Folgendes: in den neuen Datentabellen gibt es Werte für die Transportkapazität, die ich jetzt auch genutzt habe, um sie so auch im Modell zu haben. Damit haut der Assert manchmal aber nicht hin, da die Kapazität zu gering ist. Also entweder 1) wir lassen die Werte so wie vorher und machen das Assert raus und leben damit, dass die Kapazität nicht immer ausreicht oder 2) ich mache die alten Werte (ich muss die Tage nochmal nachvollziehen, wie die genau berechnet werden) und das Assert wieder rein. Was denkst du?

Reply:
@Ultramarin95  was meinst du?  Also ich denke die bessere Formulierung von Simon's Frage ist: "Ist es korrekter" wenn wir in Zukunft manchmal weniger als die benötigte Transport Kapazität berechnen? Oder zeigt das das unsere neue Berechnung irgendwo nen Fehler hat?

Reply:
Es ändert sich nur die Transportkapazität vom Flugverkehr bei allen AGS, die in der neuen Liste für Flugverkehr vorhanden sind. Ich habe mal ein bisschen geschaut, welche AGS denn so betroffen sind und das sind bisher alles AGS, die keinen Flugverkehr besitzen und dadurch die Transport Kapazität auf 0 geschrieben wird. Daher kann ich mir das nur erklären, dass in den raumtyp-spezifischen Annahmen auch Flugverkehr inbegriffen ist. Es wird in res.transport.transport_capacity_pkm auch der Flugverkehr miteinbezogen, der dann teilweise eben 0 ist. Daher schlägt der assert provided > result meiner Meinung nach fehl.

Reply:
So ganz kann ich das nicht nachvollziehen, weil meiner Meinung nach der Flugverkehr die Zahl nur größer und nicht kleiner machen sollte (also sprich das assert sollte "mehr stimmen" und nicht "weniger stimmen").

Aber wir haben einen Plan mehr zu wissen:

a) nochmal ne AGS rausfinden bei der der assert im neuen code fehlschlägt
b) nochmal überprüfen das das assert mit dem alten code und der ags nicht fehlschlägt und die result json speichern (nennen wir sie old.json) 
c) assert auskommentieren im neuen code, mit der AGS laufen lassen und json abspeichern (new.json)
d) diff old.json new.json 
e) diff ansehen und das sollte uns eigentlich zeigen welche Komponente der Berechnung schuldig ist.

Ganz am Ende dann die entsprechende AGS in unsere end_to_end tests einbauen damit der Fehler nicht wieder auftauchen kann.

Reply:
Sooo, endlich ein paar Neuigkeiten.
1) Der Assert schlägt auch in Main bei manchen AGS fehl. (siehe z.B. 01054039). Ich hatte glaube 3000 AGS getestet und 6 davon schlugen fehl.
2) Wenn ich das neue Traffic einbaue, schlagen deutlich mehr fehl
3) Es muss an der Transportkapazität des Schiffverkehrs liegen, da Flug nicht miteinbezogen wird
4) Vorher wird Schiff anhand der Bewohnereinzahl berechnet. Jetzt ist auf das tatsächliche Vorhandensein eines Hafens bezogen. Wenn dann keiner da ist, ist die Transportkapazität natürlich auf 0. Ergo verringert sich die Transportkapazität im Vergleich zur vorherigen Berechnung und der Assert schlägt ggf. fehl.
5) Es findet eben wenn ich den Traffic Ship auf das Vorhandensein eines Hafens beziehe, entkopple ich sozusagen die Benötigte (required) Transportkapazität mit der tatsächlichen in dem Sektor. Das war eben vorher nicht der Fall, weil vorher sowohl die required als auch die tatsächlich vorhandene Kapazität von der Einwohneranzahl abhängt.

Reply:
Sooo, endlich ein paar Neuigkeiten.
1) Der Assert schlägt auch in Main bei manchen AGS fehl. (siehe z.B. 01054039). Ich hatte glaube 3000 AGS getestet und 6 davon schlugen fehl.
2) Wenn ich das neue Traffic einbaue, schlagen deutlich mehr fehl
3) Es muss an der Transportkapazität des Flugverkehrs liegen, da Schiffe überhaupt keine Personentransport modelliert haben
4) In der neuen Berechnung haben Städte 0 Pkm, wenn kein Flughafen vorhanden ist. Vorher hatten eben alle Städte flughäfen inbegriffen.
5) Der Kommentar zur benötigen Transportkapazität besagt, dass keine Flughäfen inbegriffen sind. Dennoch werden im Assert die pkm für Flug miteinbezogen.
6) Bezieht man diese bei allen nicht mehr ein, schlagen deutlich mehr fehl.
7) Die Modellierung der benötigten Pkm bezieht sich auf die Einwohnerzahl, die jetzige Modelleriung der Pkm eben nicht mehr und ist von der Bewohnerzahl entkoppelt. Dadurch entsteht der Fehler.

Reply:
Ich hab mir die Berechnung der Transport Kapazitäten nun endlich genauer angeschaut. Es wird anfangs die nötige Transportkapazität anhand eines Faktors (Transportkapazität 2050 pro Person) * Bewohneranzahl berechnet. Diese notwendige Transportkapazität wird dann per Modal Split auf die verschiedenen Verkehrsträger aufgeteilt. 
Es stellte sich heraus, dass die Faktoren für Kleinstädte (SMCITY) nur 99 % der Gesamttransportkapazität abdecken, für City und Rural dafür 101 %. So ist das aber richtig der Quelle entnommen. Wsl. ein Rundungsfehler bei der Berechnung. In der "alten" Berechnung wurde das ganze meist durch den Flugverkehr ausgeglichen. Dies ist in der neuen Berechnung meist nicht mehr der Fall. Daher der Fehler. Frage ist nur, wie damit umzugehen ist. Irgendwo den 1 % herzaubern?

Die Faktoren hierzu kommen aus Agora Energiewende 2050 S.88. Siehe hierzu die detaillierte Faktorenaufschlüsselung:

City:

• Road
	• Ass_T_D_trnsprt_ppl_city_car1_frac_2050: 0.20    (Anteil Halbstadt MIV)
	• Ass_T_D_trnsprt_ppl_city_car2_frac_2050: 0.05    (Anteil Halbstadt Car Sharing)
	• Ass_T_D_trnsprt_ppl_city_car3_frac_2050: 0.05    (Anteil Halbstadt Ride Sharing und Taxis)
	• Ass_T_D_trnsprt_ppl_city_car4_frac_2050: 0.10    (Anteil Halbstadt Ride Pooling)
• ÖV
	• Ass_T_D_trnsprt_ppl_city_pt_frac_2050 : 0.46    (Anteil Halbstadt ÖV)
• Other
	• Ass_T_D_trnsprt_ppl_city_cycl_frac_2050:  0.10    (Anteil Halbstadt Fahrrad)
	• Ass_T_D_trnsprt_ppl_city_foot_frac_2050: 0.05   (Anteil Halbstadt Fuß)
• Air = Traffic New
• Ship = 0

Gesamt = 20+5+5+10+46+10+5=101 

SmallCity:

• Ass_T_D_ratio_trnsprt_ppl_to_ppl_smcity: 11237.894056847546 Pkm/Person   (Verkehrsleistung pro Person in der Halbstadt)
• Road
	• Ass_T_D_trnsprt_ppl_smcty_car1_frac_2050: 0.33    (Anteil Halbstadt MIV)
	• Ass_T_D_trnsprt_ppl_smcty_car2_frac_2050: 0.10    (Anteil Halbstadt Car Sharing)
	• Ass_T_D_trnsprt_ppl_smcty_car3_frac_2050: 0.03    (Anteil Halbstadt Ride Sharing und Taxis)
	• Ass_T_D_trnsprt_ppl_smcty_car4_frac_2050: 0.07    (Anteil Halbstadt Ride Pooling)
• ÖV: Ass_T_D_trnsprt_ppl_smcty_pt_frac_2050 : 0.36    (Anteil Halbstadt ÖV)
	• Road
		○ RoadBus
	• Rail
		○ Distance
		○ Metro
• Other
	• Ass_T_D_trnsprt_ppl_smcty_cycl_frac_2050:  0.06    (Anteil Halbstadt Fahrrad)
	• Ass_T_D_trnsprt_ppl_smcty_foot_frac_2050: 0.04    (Anteil Halbstadt Fuß)
• Air = Traffic New
• Ship = 0

Gesamt = 33 + 10 + 3 + 7 + 36 + 6 + 4 = 99  

Rural:

• Road
	○ Ass_T_D_trnsprt_ppl_rural_car1_frac_2050: 0.35    (Anteil Land MIV)
	○ Ass_T_D_trnsprt_ppl_rural_car2_frac_2050: 0.11    (Anteil Land Car Sharing)
	○ Ass_T_D_trnsprt_ppl_rural_car3_frac_2050: 0.03    (Anteil Land Ride Sharing und Taxis)
	○ Ass_T_D_trnsprt_ppl_rural_car4_frac_2050: 0.07    (Anteil Land Ride Pooling)
• ÖV 
	• Ass_T_D_trnsprt_ppl_rural_pt_frac_2050 : 0.35    (Anteil Halbstadt ÖV)
• Other
	• Ass_T_D_trnsprt_ppl_rural_cycl_frac_2050:  0.06    (Anteil Halbstadt Fahrrad)
	• Ass_T_D_trnsprt_ppl_rural_foot_frac_2050: 0.04    (Anteil Halbstadt Fuß)
• Air = Traffic New
• Ship = 0

Gesamt = 35 + 11 + 3 + 7 + 35 + 6 + 4 = 101

Reply:
Okay Hauke told me some things offline. Basically the assert should not really be there and the code should really be replaced by something quite different.  But we will if at all do this in a separate pass.